yt/utilities/lib/pixelization_routines.pyx

```diff
@@ -1520,29 +1531,31 @@ cdef np.float64_t[:] rotation_matmul(np.float64_t[:, :] rotation_matrix,
             out[i] += rotation_matrix[i, j] * coordinate_matrix[j]
     return out
 
-
 @cython.boundscheck(False)
 @cython.wraparound(False)
-cpdef np.float64_t[:, :] get_rotation_matrix(normal_vector):
+cpdef np.float64_t[:, :] get_rotation_matrix(normal_vector, final_vector):
     """ Returns a numpy rotation matrix corresponding to the
-    rotation of the given normal vector to the z-axis ([0, 0, 1]).
+    rotation of the given normal vector to the specified final_vector.
     See https://math.stackexchange.com/a/476311 although note we return the
     inverse of what's specified there.
     """
     cdef np.float64_t[:] normal_vector_np = np.array([normal_vector[0], normal_vector[1], normal_vector[2]], 
                                                      dtype='float_')
-    cdef np.float64_t[:] z_axis = np.array([0., 0., 1.], dtype='float_')
+    cdef np.float64_t[:] final_vector_np = np.array([final_vector[0], final_vector[1], final_vector[2]], 
+                                                     dtype='float_')
```

Comment:
I don't think we need these intermediate `_np`'s, I think we can do this at the input level:
```python
cpdef np.float64_t[:, :] get_rotation_matrix(np.float64_t[:] normal_vector, np.float64_t[:] final_vector):
```

Reply:
I'm confused.  You don't think I should be doing all of the intermediate normalization of these vectors in the few lines below this?

Reply:
Oh, sorry! No, it's the conversion to a memoryview,

```python
cdef np.float64_t[:] normal_vector_np = np.array([normal_vector[0], normal_vector[1], normal_vector[2]], dtype='float_')
```
If you change the function declaration to 
```python
cpdef np.float64_t[:, :] get_rotation_matrix(np.float64_t[:] normal_vector, np.float64_t[:] final_vector)
```
I don't _think_ you need to do that intermediate conversion step because you are passing in a  numpy array. I think with the above function declaration you can just use the `normal_vector` in place of `normal_vector_np`.

Reply:
The code definitely looks better when I make your suggestion, as it is more succinct, but unfortunately, cython gives a compile error, so I don't think this is a feasible change.

Reply:
oh fair enough. I did quickly test the following locally,

```python
@cython.boundscheck(False)
@cython.wraparound(False)
cpdef np.float64_t[:, :] get_rotation_matrix(np.float64_t[:] normal_vector,
                                             np.float64_t[:] final_vector):
    """ Returns a numpy rotation matrix corresponding to the
    rotation of the given normal vector to the specified final_vector.
    See https://math.stackexchange.com/a/476311 although note we return the
    inverse of what's specified there.
    """
    cdef np.float64_t[:] normal_unit_vector = normal_vector / np.linalg.norm(normal_vector)
    cdef np.float64_t[:] final_unit_vector = final_vector / np.linalg.norm(final_vector)
```

and didn't get any errors.

Reply:
Weird, it works this time.  I don't know what I did wrong last time.  OK, thanks.